src/Components/CreateImageWizardV2/utilities/requestMapper.tsx

```diff
@@ -0,0 +1,150 @@
+import { store } from '../../../store';
+import {
+  AwsUploadRequestOptions,
+  CreateBlueprintRequest,
+  Customizations,
+  GcpUploadRequestOptions,
+  ImageRequest,
+  ImageTypes,
+  Subscription,
+  UploadTypes,
+} from '../../../store/imageBuilderApi';
+import { wizardState } from '../../../store/wizardSlice';
+
+/**
+ * This function maps the wizard state to a valid CreateBlueprint request object
+ * @param {string} orgID organization ID
+ * @returns {CreateBlueprintRequest} creation blueprint request payload
+ */
+export const mapRequestFromState = (orgID: string): CreateBlueprintRequest => {
+  const { wizard: wizardState } = store.getState();
+  const images = getImageRequest(wizardState);
+  const customizations = getCustomizations(wizardState, orgID);
+
+  return {
+    name: wizardState.details.blueprintName,
+    description: wizardState.details.blueprintDescription,
+    distribution: wizardState.distribution,
+    image_requests: images,
+    customizations,
+  };
+};
+
+const getImageRequest = (state: wizardState): ImageRequest[] => {
+  return state.imageTypes.map((type) => ({
+    architecture: state.architecture,
+    image_type: type,
+    upload_request: {
+      type: uploadTypeByTargetEnv(type),
+      options: getImageOptions(type, state),
+    },
+  }));
+};
+
+const uploadTypeByTargetEnv = (imageType: ImageTypes): UploadTypes => {
+  switch (imageType) {
+    // TODO: add edge
+    case 'aws':
+      return 'aws';
+    case 'gcp':
+      return 'gcp';
+    case 'azure':
+      return 'azure';
+    case 'oci':
+      return 'oci.objectstorage';
+    case 'wsl':
+      return 'aws.s3';
+    case 'image-installer':
+      return 'aws.s3';
+    case 'vsphere':
+      return 'aws.s3';
+    case 'ami':
+      return 'aws';
+    default: {
```

Comment:
This might  happen. because edge images haven't implemented yet, returning a string is forced by the image builder API, I'd prefer to return undefined and raise an error.  The wizard is in progress work under experimental flag, there's a TODO for the missing `edge` types, I think throwing an error would be better than just a warning 👍